### ISSUE TITLE
Deployment slots with Functions

### ISSUE THREAD

**Issue by lindydonna** [Azure/Azure-Functions#45]
Article explaining how to use it with Azure Functions

**Comment by miantosca** [comment]
Are deployment slots planned for azure functions? Is the alternative for now to setup a function app service for each slot - dev, test, prod for instance and deploy to those?

**Comment by lindydonna** [comment]
They are planned, yes, but we don't have a timeline to share. A separate Function App for each slot is a good solution in the meantime.

**Comment by SimonLuckenuik** [comment]
@lindydonna, is the limitation applicable to both Dedicated and Consumption Plans, or only Consumption Plan?

**Comment by SimonLuckenuik** [comment]
With the newly released Functions Proxies, using the Application Settings to customize the URL could be a good temporary alternative to the deployment slots:  https://%CURRENT_FUNCTION_DOMAIN%/api/orders

Ref: https://docs.microsoft.com/en-us/azure/azure-functions/functions-proxies#a-namemodify-requestsamodifying-backend-requests

**Comment by lindydonna** [comment]
@SimonLuckenuik The limitation is on both regular App Service and Consumption plans. 

That is a really nice trick on using Functions Proxies!

**Comment by SteveALee** [comment]
What is the best way to escalate this as a feature request?

**Comment by lindydonna** [comment]
@SteveALee You should consider it already escalated as a feature request, and it's under development. (No dates to share, unfortunately.)

**Comment by APErebus** [comment]
ooooo, what is this? 
![slots?](https://cloud.githubusercontent.com/assets/332730/25832970/7fe5aa8a-34b1-11e7-8ffd-996a1471ff3f.png)

**Comment by fabiocav** [comment]
😀

**Comment by kverpoorten** [comment]
I'd like to deploy my function app to a slot using a VSTS build and release. Are there any manuals available for that already?

**Comment by sjkp** [comment]
@kverpoorten as far as I can see there is nothing special about the slots from webapp slots, so you should be able to use the same vsts task to deploy to an azure function slot as you use to deploy to an web app slot.

**Comment by sadgit** [comment]
I have enabled a develop slot and tried to switch it to the develop branch - now it seems FUBAR on both the slot and production.

**Comment by fabiocav** [comment]
@sadgit can you provide more information about what you're seeing? What error(s) are you running into?

**Comment by sadgit** [comment]
I have several Azure function apps with CI from VisualStudio.com. With the release of Slots setup a 'develop' slot for one branch and then production slot for Master. Setup the CI on the slot to take from 'develop'. The slot simply shows No Deployments.
An afternoon of head scratching and frantic Binging I found the history tab and these errors. This is now happening on other function apps without slots.

**Comment by sadgit** [comment]
PS - I wondered whether it was something the cloning of the slot during slot creation process?
I seems a common use case to have Master to Production perhaps with auto-swap and a develop version. Same repo different branch.

**Comment by sadgit** [comment]
Hooray - all seems to be back working this morning
For logs check here
https://**<myCorp>**.visualstudio.com/**<myProj>**/_apps/hub/ms.vss-servicehooks-web.manageServiceHooks-project

Thanks MSFT

**Comment by napalm684** [comment]
Is there a reason why deployment slots are not compatible with the consumption plan?  Is this a short term limitation or going to be a long term restriction?

**Comment by lindydonna** [comment]
@napalm684 Deployment slots are supported for the consumption plan; is it not working for you?

**Comment by napalm684** [comment]
@lindydonna correct it gives me the error that only one slot is supported for the consumption plan if I need multiple I need to use an app service plan.

**Comment by lindydonna** [comment]
@napalm684 That's correct, only one slot is currently supported on the Consumption plan. We may change this in the future.

**Comment by napalm684** [comment]
@lindydonna I hope more get added a single slot really doesn't work for me under the premise of trying to consolidate down to a single function app with slots for dev, test, stage and production.  Another related question, if you use slots can FTPs be directed to deploy to a specific slot?  If that doesn't work then slots just aren't going to cover my scenario currently anyway.

**Comment by PingPongSet** [comment]
wathing it

**Comment by petro2050** [comment]
I was able to start using deployment slots on a consumption plan, but I'm having issues with the auto swap feature. I'm getting 503 errors for a few seconds when the new deployment becomes active.

https://github.com/Azure/Azure-Functions/issues/1655
